app/controllers/recipients_controller.rb

```diff
@@ -0,0 +1,41 @@
+class RecipientsController < ApplicationController
```

Comment:
since most of the logic below is operating upon a `message` object, this code should probably be moved to the `MessagesController`.  By convention, each controller should apply to it's referenced model.